pkg/intermediate/aggregate.go

```diff
@@ -110,15 +110,26 @@ func (a *AggregationProcess) AggregateMsgByFlowKey(message *entities.Message) er
 		return nil
 	}
 	records := set.GetRecords()
+	invalidRecs := 0
 	for _, record := range records {
-		flowKey, err := getFlowKeyFromRecord(record)
-		if err != nil {
-			return err
-		}
-		if err = a.addOrUpdateRecordInMap(flowKey, record); err != nil {
-			return err
+		// Validate the data record. If invalid, we log the error and move to the next
+		// record.
+		if !validateDataRecord(record) {
+			klog.V(2).Infof("Invalid data record because decoded values of elements are not valid.")
```

Comment:
seems it is more an `Errorf` than `Infof`

Reply:
Yes `Errorf` is more appropriate. The intention was to use error log but missed it.